safe_control_gym/controllers/mpc/linear_mpc.py

```diff
@@ -165,7 +157,7 @@ def setup_optimizer(self):
                           u=np.zeros((nu, 1)) + self.U_EQ[:, None],
                           Xr=x_ref[:, -1],
                           Ur=np.zeros((nu, 1)),
-                          Q=self.Q,
+                          Q=self.Q if not self.use_lqr_gain_and_terminal_cost else self.P,
```

Comment:
would be interested in learning more about this

Reply:
This is basically using the LQR solution for 1) terminal cost and 2) ancillary gain and to improve the MPC solution quality. But since the terminal constraint set is missing, this wouldn't provide any rigorous theoretical guarantee.